app/app_test.go

```diff
@@ -63,6 +69,64 @@ func TestExport(t *testing.T) {
 	assert.Len(t, exportedApp.Validators, 1) // no validators set in default genesis
 }
 
+// TestLegacyMsgAreAminoRegistered checks if all known msg types are registered on the app's amino codec.
+// It doesn't check if they are registered on the module codecs used for signature checking.
+func TestLegacyMsgAreAminoRegistered(t *testing.T) {
+	tApp := NewTestApp()
+
+	lcdc := tApp.LegacyAmino()
+
+	// Use the proto codec as the canonical list of msg types.
+	protoCodec := tApp.AppCodec().(*codec.ProtoCodec)
+	msgProtoNames := protoCodec.InterfaceRegistry().ListImplementations(sdk.MsgInterfaceProtoName)
+
+	for i, msgName := range msgProtoNames {
+		// skip external unregistered msgs
+		if msgName == sdk.MsgTypeURL(&evmtypes.MsgEthereumTx{}) ||
```

Comment:
i'd love if this was a list called `unregisteredMsgs` or something similar

Reply:
agreed, it's a bit confusing, changed to `protoRegisteredMsgs` as at this point it's unclear if these msgs are all registered on amino or not